standard/expressions.md

```diff
@@ -737,7 +757,8 @@ For each of the method arguments `Eᵢ`:
 
 - If `Eᵢ` is an anonymous function, an *explicit parameter type inference* ([§12.6.3.8](expressions.md#12638-explicit-parameter-type-inferences)) is made *from* `Eᵢ` *to* `Tᵢ`
 - Otherwise, if `Eᵢ` has a type `U` and `xᵢ` is a value parameter ([§15.6.2.2](classes.md#15622-value-parameters)) then a *lower-bound inference* ([§12.6.3.10](expressions.md#126310-lower-bound-inferences)) is made *from* `U` *to* `Tᵢ`.
-- Otherwise, if `Eᵢ` has a type `U` and `xᵢ` is a reference ([§15.6.2.3](classes.md#15623-reference-parameters)) or output ([§15.6.2.4](classes.md#15624-output-parameters)) parameter then an *exact inference* ([§12.6.3.9](expressions.md#12639-exact-inferences)) is made *from* `U` *to* `Tᵢ`.
+- Otherwise, if `Eᵢ` has a type `U` and `xᵢ` is a reference parameter ([§15.6.2.3](classes.md#15623-reference-parameters)), or output parameter ([§15.6.2.4](classes.md#15624-output-parameters)) then an *exact inference* ([§12.6.3.9](expressions.md#12639-exact-inferences)) is made *from* `U` *to* `Tᵢ`.
+- Otherwise, if `Eᵢ` has a type `U` and `xᵢ` is an input parameter (§method-input-parameters-new-clause) then a *lower bound inference* ([§12.6.3.10](expressions.md#126310-lower-bound-inferences)) is made *from* `U` *to* `Tᵢ`. A new storage location is created following the rules of §12.6.2.2 on corresponding parameters.
```

Comment:
>A new storage location is created following the rules of §12.6.2.2 on corresponding parameters. [](http://example.com/codeflow?start=227&length=95)

Should this be part of the invocation description rather than type inference?

Reply:
I think it's needed here. The purpose is to allow implicit conversions to an `in` parameter when the argument doesn't include the `in` modifier. For example:

```csharp
public void M(in long p) {}

// calling:

long num = 25;
int i = 19;

M(in i); // Not allowed. Implicit conversion, not an identity conversion.
M (i); // OK, implicit conversion and the creation of temporary.
M(in num); // OK
M(num); // OK.

Reply:
Does the creation of a storage location affect type inference though?

Reply:
Missing description of exact inference from an input `in` argument to an `in` parameter, which requires exact inference.